File: code/test_suites/cpp/FunctionBuildTest/tests/test_Polars_filter.cpp

#include <gtest/gtest.h>
#include "../src/function_Polars_filter.cpp"

class TestPolarsExample : public ::testing::Test {
protected:
    PolarsExample calc;
};

TEST_F(TestPolarsExample, TestFilterNumbersGreaterThanThreshold) {
    auto result = calc.filter_numbers(2, 1, 2, 3);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 3.0f);
}

TEST_F(TestPolarsExample, TestFilterNumbersMultipleValues) {
    auto result = calc.filter_numbers(1, 5, 2, 7);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 7.0f);
}

TEST_F(TestPolarsExample, TestFilterNumbersEmptyResult) {
    auto result = calc.filter_numbers(10, 1, 2, 3);
    EXPECT_FALSE(result);
}

TEST_F(TestPolarsExample, TestFilterNumbersSingleValueGreaterThanThreshold) {
    auto result = calc.filter_numbers(0, 1);
    ASSERT_TRUE(result);
    EXPECT_FLOAT_EQ(*result, 1.0f);
}

TEST_F(TestPolarsExample, TestFilterNumbersNoValuesAboveThreshold) {
    auto result = calc.filter_numbers(100, 10, 20, 30);
    EXPECT_FALSE(result);
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/data/python/tests/test_Polars_filter.py
import unittest
from function_Polars_filter import PolarsExample

class TestPolarsExample(unittest.TestCase):

    def test_filter_numbers_greater_than_threshold(self):
        calc = PolarsExample()
        result = calc.filter_numbers(2, 1, 2, 3)
        self.assertEqual(result, 3.0)

    def test_filter_numbers_multiple_values(self):
        calc = PolarsExample()
        result = calc.filter_numbers(1, 5, 2, 7)
        self.assertEqual(result, 7.0)

    def test_filter_numbers_empty_result(self):
        calc = PolarsExample()
        result = calc.filter_numbers(10, 1, 2, 3)
        self.assertIsNone(result)

    def test_filter_numbers_single_value_greater_than_threshold(self):
        calc = PolarsExample()
        result = calc.filter_numbers(0, 1)
        self.assertEqual(result, 1.0)

    def test_filter_numbers_no_values_above_threshold(self):
        calc = PolarsExample()
        result = calc.filter_numbers(100, 10, 20, 30)
        self.assertIsNone(result)

if __name__ == '__main__':
    unittest.main()
